packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -111,8 +111,16 @@ class IosProject extends XcodeBasedProject {
   @override
   String get pluginConfigKey => IOSPlugin.kConfigKey;
 
-  static final RegExp _productBundleIdPattern = RegExp(r'''^\s*PRODUCT_BUNDLE_IDENTIFIER\s*=\s*(["']?)(.*?)\1;\s*$''');
-  static const String _productBundleIdVariable = r'$(PRODUCT_BUNDLE_IDENTIFIER)';
+  // build setting keys
+  static const String kProductBundleIdKey = 'PRODUCT_BUNDLE_IDENTIFIER';
+  static const String kTeamIdKey = 'DEVELOPMENT_TEAM';
+  static const String kEntitlementFilePathKey = 'CODE_SIGN_ENTITLEMENTS';
+  static const String kHostAppBundleNameKey = 'FULL_PRODUCT_NAME';
+
+  static final RegExp _productBundleIdPattern = RegExp('^\\s*$kProductBundleIdKey\\s*=\\s*(["\']?)(.*?)\\1;\\s*\$');
+  static const String _kProductBundleIdVariable = '\$($kProductBundleIdKey)';
+
+  static final RegExp _associatedDomainPattern = RegExp(r'''^applinks:(.*)''');
```

Comment:
I don't think you need triple quotes

```suggestion
  static final RegExp _associatedDomainPattern = RegExp(r'^applinks:(.*)');
```